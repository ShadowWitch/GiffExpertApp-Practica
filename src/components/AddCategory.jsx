import { useState } from "react"
import {PropTypes} from 'prop-types'

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('')

    const onSubmit = (e) =>{
        e.preventDefault()
        // console.log('ACA >>> ', inputValue)
        // console.log('ACA >>', e.target.value);

        if(inputValue.trim().length <= 1) return;
        setInputValue('')
        onNewCategory(inputValue)
    }

    const onInputChange = ({target}) =>{
        // console.log('TARGET ACA>> ', target.value)
        // console.log(target)
        setInputValue(target.value)
    }

    return (
        <form
            // className="form_gifs"
            onSubmit={onSubmit}
            aria-label="form" // "aria-label" para que luego lo podamos llamar desde los Testing con "getByRole('form')" ya que luego no lo reconocera si no se lo pongo
        >
            <input 
                type="text" 
                placeholder="Seatch Gif"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}