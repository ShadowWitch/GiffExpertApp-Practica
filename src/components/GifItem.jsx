
import {PropTypes} from 'prop-types';

export const GifItem = ({title, url}) => {
    // console.log(url);
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

// Config "prop-types"
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

// Specifies the default values for props:
// GifItem.defaultProps = {
//     title: 'Image Not Found',
//     url: 'https://media0.giphy.com/media/lqFHf5fYMSuKcSOJph/giphy.gif?cid=284aee55m6ffwyzjtf0gogvv8ruy5aicg4gr0x9gii75v8v8&rid=giphy.gif&ct=g'
//   };
  