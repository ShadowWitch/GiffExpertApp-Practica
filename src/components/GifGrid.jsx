
import { GifItem } from './GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types';

// https://api.giphy.com/v1/gifs/search?api_key=uGrjbURJnoZDKb5xysNp2bUD8ayZqDw7&q=OnePiece&limit=16

export const GifGrid = ({category}) =>{

    const {images, isLoading} = useFetchGifs(category)
    // console.log(category)
    // console.log(typeof category);

    return (
        <>
            <div>
                <h3>{category}</h3>
                {/* {
                    isLoading ? (<h2>Cargando...</h2>) : null
                } */}
                {
                    isLoading && (<h2>Cargando...</h2>) // Si estra en "true"
                }
                <div className='card-grid'>
                    {
                        images.map( image => (
                            <GifItem 
                                key={image.id}
                                // image={image}
                                {...image}
                            />
                        ))
                    }
                </div>
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
