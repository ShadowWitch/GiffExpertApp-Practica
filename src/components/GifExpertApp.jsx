import { useState } from "react"
import { AddCategory } from "./AddCategory"
import { GifGrid } from "./GifGrid"

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Naruto'])

    const onAddCategory = (newCategory) => {
        if(categories.includes(newCategory)) return alert('Ya existe');
        setCategories([...categories, newCategory])
        // setCategories(['Hola mundo', ...categories])
        // setCategories(cat => [...cat, 'Hola mundo'])
    }

    const arr1 = [1, 2, 3]

    return (
        <>
            <h1>GifExpert</h1>

            <AddCategory 
                onNewCategory={onAddCategory}
            />

            {
                categories.map(category => (
                        <GifGrid 
                            category={category}
                            key={category}
                        />
                    ))
            }

        </>
    )
}
