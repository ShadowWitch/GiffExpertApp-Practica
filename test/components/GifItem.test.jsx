import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem />', () => {
    const title = 'One Punch'
    const url = 'https://media0.giphy.com/media/lqFHf5fYMSuKcSOJph/giphy.gif?cid=284aee55m6ffwyzjtf0gogvv8ruy5aicg4gr0x9gii75v8v8&rid=giphy.gif&ct=g'
    test('Debe de hacer match con el SnapShot', () =>{

        // const {container} = render(<GifItem title={title} url={url} />)
        // console.log(container)

        const {container} = render(<GifItem title={title} url={url} />)
        // console.log(container)

        expect(container).toMatchSnapshot();
    })

    test('Debe mostrar la imagen con el URL y la Alt indicada', () =>{

        render(<GifItem title={title} url={url} />)
        // screen.debug(); // ver componente renderizado en memoria (para recordar como va)
        // expect(screen.getByRole("img").src).toBe(url)
        // expect(screen.getByRole("img").alt).toBe(title)

        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('Debe mostrar el titulo en un componente', () => {
        render(<GifItem title={title} url={url} />)
        // console.log(screen.getByText(title)) // Buscara en el componente donde este la palabra que tiene "title"
        // console.log(screen.getByText(title))

        expect(screen.getByText(title)).toBeTruthy() // "toBeTruthy()" Comprueba si existe
    })
})

// qwe
