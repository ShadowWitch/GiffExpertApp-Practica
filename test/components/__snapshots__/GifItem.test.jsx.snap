// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el SnapShot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://media0.giphy.com/media/lqFHf5fYMSuKcSOJph/giphy.gif?cid=284aee55m6ffwyzjtf0gogvv8ruy5aicg4gr0x9gii75v8v8&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
